pallets/liquid-staking-v2/src/lib.rs

```diff
@@ -311,6 +339,75 @@ mod pallet {
             Ok(().into())
         }
 
+        /// claim staking currency for given (owner, era_index)
+        ///
+        /// - `owner`: unstaker
+        /// - `era_index`: era_index in which user unstaked
+        #[pallet::weight(T::WeightInfo::claim())]
+        #[transactional]
+        pub fn claim(
+            origin: OriginFor<T>,
+            owner: T::AccountId,
+            era_index: EraIndex,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            Self::ensure_era_finished(era_index)?;
+
+            let unstake_misc = Self::account_unstake(&owner, era_index);
+            let matching_ledger = Self::matching_pool(era_index);
+            if unstake_misc.total_amount.is_zero()
+                || unstake_misc.claimed_amount >= unstake_misc.total_amount
+            {
+                return Err(Error::<T>::NothingToClaim.into());
+            }
+
+            let current_era = Self::current_era();
+
+            // TODO check if asset has been transferred back from relaychain
+            // passed bonding duration & assets has been withdrawn to pallet account
```

Comment:
You check the withdraw unbond state in `StakingOperationHistory` to address this TODO.